collector-bitmovin-player/src/main/java/com/bitmovin/analytics/bitmovin/player/BitmovinSdkAdapter.java

```diff
@@ -94,163 +94,182 @@ public void init() {
-        this.bitmovinPlayer.removeEventListener(onErrorListener);
-        this.bitmovinPlayer.removeEventListener(onSubtitleChangedListener);
-        this.bitmovinPlayer.removeEventListener(onAudioChangedListener);
-        this.bitmovinPlayer.removeEventListener(onDownloadFinishedListener);
-        this.bitmovinPlayer.removeEventListener(onDestroyedListener);
-
-        this.bitmovinPlayer.removeEventListener(onAdBreakStartedListener);
-        this.bitmovinPlayer.removeEventListener(onAdBreakFinishedListener);
+        try {
+            Log.d(TAG, "Removing Player Listeners");
+            this.bitmovinPlayer.removeEventListener(onSourceLoadedListener);
+            this.bitmovinPlayer.removeEventListener(onSourceUnloadedListener);
+
+            this.bitmovinPlayer.removeEventListener(onPlayListener);
+            this.bitmovinPlayer.removeEventListener(onPlayingListener);
+            this.bitmovinPlayer.removeEventListener(onPausedListener);
+            this.bitmovinPlayer.removeEventListener(onStallEndedListener);
+            this.bitmovinPlayer.removeEventListener(onSeekedListener);
+            this.bitmovinPlayer.removeEventListener(onStallStartedListener);
+            this.bitmovinPlayer.removeEventListener(onSeekListener);
+            this.bitmovinPlayer.removeEventListener(onPlaybackFinishedListener);
+            this.bitmovinPlayer.removeEventListener(onReadyListener);
+            this.bitmovinPlayer.removeEventListener(onVideoPlaybackQualityChangedListener);
+            this.bitmovinPlayer.removeEventListener(onAudioPlaybackQualityChangedListener);
+            this.bitmovinPlayer.removeEventListener(onDroppedVideoFramesListener);
+            this.bitmovinPlayer.removeEventListener(onErrorListener);
+            this.bitmovinPlayer.removeEventListener(onSubtitleChangedListener);
+            this.bitmovinPlayer.removeEventListener(onAudioChangedListener);
+            this.bitmovinPlayer.removeEventListener(onDownloadFinishedListener);
+            this.bitmovinPlayer.removeEventListener(onDestroyedListener);
+
+            this.bitmovinPlayer.removeEventListener(onAdBreakStartedListener);
+            this.bitmovinPlayer.removeEventListener(onAdBreakFinishedListener);
+        }
+        catch (Exception e) {
+            Log.d(TAG, "On Removing Player Listeners", e);
+        }
     }
 
     @Override
     public EventData createEventData() {
-        EventData data = factory.build(stateMachine.getImpressionId());
-
-        data.setAnalyticsVersion(BuildConfig.VERSION_NAME);
-        data.setPlayer(PlayerType.BITMOVIN.toString());
-
-        //duration
-        double duration = bitmovinPlayer.getDuration();
-        if (duration != Double.POSITIVE_INFINITY) {
-            data.setVideoDuration((long) duration * Util.MILLISECONDS_IN_SECONDS);
-        }
-
-        //ad
-        if (bitmovinPlayer.isAd()) {
-            data.setAd(1);
-        }
+        try {
```

Comment:
Please remove here as well: This is Java code - we never expect something to be non-nullable here (check the line with video-quality for example: we get the object, and then check if it's null or not, but never pretend that it's non-nullable).
We need to make sure to also do that in future in our Java classes, and at one point, when we change to Kotlin, those things should be breaking changes in the player code. But I'd prefer this to crash then as well.
The reason why we fix this now can be seen in our Ads-code, as we use Kotlin there: (just an example, not sure if this was the exact reason:) Older player versions have the `AdBreakEvent` with a non-nullable `AdBreak`, which will translate to `val adBreak: AdBreak` in Kotlin. So in our auto-complete etc., we would directly access it, because obviously it cannot be null. Player-team introduced a change though, where they changed this to `val adBreak: AdBreak?`. So if you update the player, the code actually still works, because both would translate to `public AdBreak getAdBreak()` in Java - but in Java it can return `null`. This is where our code would crash if a newer player-version would return `null`.
So for now, let's just keep those try-catch blocks to the eventhandlers, not to everything IMO.
I'd like to have @Tigraine 's opinion as well though, before you change anything.